dali/pipeline/operators/fused/crop_mirror_normalize.cc

```diff
@@ -118,9 +118,31 @@ void CropMirrorNormalize<CPUBackend>::RunHelper(SampleWorkspace *ws,
   const int mirror_image = !has_mirror_ ? mirror_.data<int>()[0] :
      spec_.GetArgument<int>("mirror", ws, ws->data_idx());
 
+  vector<Index> input_shape = input.shape();
+  DALI_ENFORCE(input_shape.size() == 3,
+      "Expects 3-dimensional image input.");
+
+  int H = input_shape[0];
+  int W = input_shape[1];
+  // Crop
+  DALI_ENFORCE(H >= crop_h_);
+  DALI_ENFORCE(W >= crop_w_);
+
+  float crop_x_image_coord = spec_.GetArgument<float>("crop_pos_x", ws, ws->data_idx());
+  float crop_y_image_coord = spec_.GetArgument<float>("crop_pos_y", ws, ws->data_idx());
+
+  DALI_ENFORCE(crop_x_image_coord >= 0.f && crop_x_image_coord <= 1.f,
+      "Crop coordinates need to be in range [0.0, 1.0]");
+  DALI_ENFORCE(crop_y_image_coord >= 0.f && crop_y_image_coord <= 1.f,
+      "Crop coordinates need to be in range [0.0, 1.0]");
+
+  int crop_y = crop_y_image_coord * (H - crop_h_);
```

Comment:
The distribution is skewed. Should be:
```suggestion
  int crop_y = std::floor(crop_y_image_coord * (H - crop_h_) + 0.5f);
```

Reply:
Will do (for GPU and crop itself as well).

Reply:
Done